runtime/src/main/java/org/corfudb/runtime/collections/streaming/StreamingTask.java

```diff
@@ -76,20 +77,38 @@ public StreamingTask(CorfuRuntime runtime, ExecutorService workerPool, String na
         this.listenerId = String.format("listener_%s_%s_%s", listener, namespace, streamTag);
         this.listener = listener;
         TableRegistry registry = runtime.getTableRegistry();
-        final UUID streamId = TableRegistry.getStreamIdForStreamTag(namespace, streamTag);
+        final UUID streamId;
+        // Federated tables do not have a stream tag, only an option isFederated set to true. Internally,
+        // their stream tag is constructed when the table is opened. This stream tag is constructed differently than
+        // other tables. So for subscribers on this tag, do not construct the streamId using the generic algorithm.
+        if (streamTag.equals(ObjectsView.getLogReplicatorStreamId().toString())) {
+            streamId = ObjectsView.getLogReplicatorStreamId();
+        } else {
+            streamId = TableRegistry.getStreamIdForStreamTag(namespace, streamTag);
+        }
         this.stream = new DeltaStream(runtime.getAddressSpaceView(), streamId, address, bufferSize);
         this.tableSchemas = tablesOfInterest
                 .stream()
                 .collect(Collectors.toMap(
                         tName -> CorfuRuntime.getStreamID(TableRegistry.getFullyQualifiedTableName(namespace, tName)),
                         tName -> {
-                            // The table should be opened with full schema before subscription.
-                            Table<K, V, M> t = registry.getTable(namespace, tName);
-                            if (!t.getStreamTags().contains(streamId)) {
-                                throw new IllegalArgumentException(String.format("Interested table: %s does not " +
+                            // Subscription on special tables(ProtobufDescriptor and Registry tables) is not possible
+                            // with the regular workflow as they are not opened using CorfuStore.openTable().
+                            // However, these tables do have the stream tag for Log Replication and it is useful to
+                            // subscribe to them for testing purposes. The below is special handling to allow for such a
+                            // subscription.
+                            if (streamId.equals(ObjectsView.getLogReplicatorStreamId()) && namespace.equals
+                                (TableRegistry.CORFU_SYSTEM_NAMESPACE) && tName.equals(TableRegistry.PROTOBUF_DESCRIPTOR_TABLE_NAME)) {
```

Comment:
In the comment you mention descriptor table and registry table, however, you only treat descriptor table.